Assets/HoloToolkit-Tests/Input/Scripts/DictationRecordButton.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity.InputModule.Tests
+{
+    public class DictationRecordButton : MonoBehaviour, IInputClickHandler, IDictationHandler
+    {
+        [SerializeField]
+        [Range(0.1f, 5f)]
+        [Tooltip("The time length in seconds before dictation recognizer session ends due to lack of audio input in case there was no audio heard in the current session.")]
+        private float initialSilenceTimeout = 5f;
+
+        [SerializeField]
+        [Range(5f, 60f)]
+        [Tooltip("The time length in seconds before dictation recognizer session ends due to lack of audio input.")]
+        private float autoSilenceTimeout = 20f;
+
+        [SerializeField]
+        [Range(1, 60)]
+        [Tooltip("Length in seconds for the manager to listen.")]
+        private int recordingTime = 10;
+
+        [SerializeField]
+        private TextMesh speechToTextOutput;
+
+        [SerializeField]
+        private GameObject recordLight;
+
+        private Renderer buttonRenderer;
+
+        private bool isRecording;
+
+        private void Awake()
+        {
+            buttonRenderer = GetComponent<Renderer>();
+            DictationInputManager.InitialSilenceTimeout = initialSilenceTimeout;
+            DictationInputManager.AutoSilenceTimeout = autoSilenceTimeout;
+            DictationInputManager.RecordingTime = recordingTime;
+        }
+
+        public void OnInputClicked(InputClickedEventData eventData)
+        {
+            ToggleRecording();
+        }
+
+        private void ToggleRecording()
+        {
+            if (isRecording)
+            {
+                isRecording = false;
+                StartCoroutine(DictationInputManager.StopRecording());
+                speechToTextOutput.color = Color.white;
+                buttonRenderer.enabled = true;
+                recordLight.SetActive(false);
+            }
+            else
+            {
+                isRecording = true;
+                StartCoroutine(DictationInputManager.StartRecording());
```

Comment:
What would happen if ToggleRecording is called to rapidly turn recording on and off? I'm concerned that StartRecording and StopRecording coroutines would both be running simultaneously--possibly even multiple instances of them. In edge cases, it seems like this could leave the system in an inconsistent state.

Not sure of the best way to handle this. 

Ideas:
- Each coroutine yields at the beginning to wait for a previous coroutine to exit before starting its work.
- There is only a single coroutine that is always running to handle starting and shutting down. A variable is set telling it the desired state (Running/Stopped) and it appropriately handles transitions whenever it detects that it isn't in the desired state. Usually, when already in the desired state, this coroutine would simply yield and do no work.